src/App/Controller/SecurityController.php

```diff
@@ -4,28 +4,84 @@
 
 namespace App\Controller;
 
+use App\Security\User;
+use L3l0Labs\SystemAccess\Email;
+use L3l0Labs\SystemAccess\UseCase\RegisterUser\Command as RegisterUserCommand;
+use App\Form\RegisterUserType;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
 use Symfony\Component\Security\Http\Authentication\AuthenticationUtils;
 
 class SecurityController extends Controller
 {
     public function login(Request $request, AuthenticationUtils $authUtils): Response
     {
-        // get the login error if there is one
         $error = $authUtils->getLastAuthenticationError();
 
-        // last username entered by the user
         $lastUsername = $authUtils->getLastUsername();
 
-        return $this->render('security/login.html.twig', array(
+        return $this->render('security/login.html.twig', [
             'last_username' => $lastUsername,
-            'error'         => $error,
-        ));
+            'error'         => $error
+        ]);
     }
 
     public function logout()
     {
     }
+
+    public function register(Request $request)
+    {
+        $form = $this->createForm(RegisterUserType::class);
+        $form->handleRequest($request);
+
+
+
+        if($form->isValid()) {
+            var_dump($form->getErrors()->current());die;
+            $formData = $form->getData();
+            $registerUserCommand = new RegisterUserCommand(
+                $formData['email'],
+                $formData['password'],
+                $formData['vatIdNumber']
+            );
+
+            $this->registerUsers()->execute($registerUserCommand);
+
+            $this->authenticateUser(new Email($formData['email']));
+
+            return $this->redirectToRoute('index');
+        }
+
+        return $this->render('security/register.html.twig', [
+            'form' => $form->createView()
+        ]);
+    }
+
+    private function registerUsers()
```

Comment:
If you extract that method already you should use return type hint. And probably that method should be called `registerUser() : RegisterUser`. I am not even sure if we need such methods.